### PR TITLE
Add error handling to db_test examples.

### PR DESCRIPTION
**Pull request by bdarnell**
Avoid using String(), which makes for convenient one-liners in this test
but is not representative of real-world use.

### CODE REVIEW

**Comment by bdarnell** [comment]
Without the one-liners using String(), the new API is a lot less compelling; we may want to think about this some more before adopting it everywhere. I don't like that the results of Get() with a single key need to be accessed via `Rows[0]` and that we have both the ability to pass multiple keys to Get() (resulting in one Result with multiple Rows) and multiple Gets in a Batch (resulting in multiple Results with one Row each).

Error handling in this interface is awkward for several reasons:
* Returning an error inside the Result struct means that errcheck can't see it
* If you type `result.Error` instead of `result.Err`, you get a function object which is never nil
* `Result` implements `error`, so an assignment like `err = db.Put(...)` will compile (assuming an existing variable of type `error`). The error will always be non-nil, but its `Error` method will panic.
* What is the relationship between the error returned by `db.Run(batch)` and the errors in the individual results? Must the app check each operation for failure in addition to the top-level error, or would this be redundant?

I think that we should remove the Err field and Error method from result, and everything that now returns a Result should return a `Result, error` pair.

File: client/db_test.go [hunk 1]
```diff
@@ -38,55 +38,104 @@ func ExampleDB_Get() {
 	s, db := setup()
 	defer s.Stop()
 
-	fmt.Println(db.Get("aa").String())
+	result := db.Get("aa")
+	if result.Err != nil {
+		panic(result.Err)
+	}
+	fmt.Printf("aa=%s\n", result.Rows[0].ValueBytes())
```
Comment (vivekmenezes) [inline client/db_test.go]: Why did you change this? You can still use .String()

Comment (bdarnell) [inline client/db_test.go]: You can, but real code wouldn't. Real code would want the value of the "aa" record, not a string like "0: aa=nil".

Comment (vivekmenezes) [inline client/db_test.go]: Ben, See the PR I just sent out that uses the new API in the banking app.
You can comment there if you like

**Comment by petermattis** [comment]
Removing `Result.Error()` makes sense. It did not work as well as expected. Removing `Result.Err` is trickier. What would you do with the `Batch` methods which want to return a `*Batch` but might generate an error if an invalid argument was passed? Perhaps leave `Result.Err`, but have the various methods return `Result, error` where the second return value is always `Result.Err`. 

Adding all of that error checking to the examples adds quite a bit of ugliness.

**Comment by tschottdorf** [comment]
I've pitched this a couple of times to no avail, but
```
result := db.Get(...)
if val, err := result.ValueBytes(); err != nil {
   ...
}
```
where `ValueBytes()` is sugar for `ValueBytes(0)`.

**Comment by petermattis** [comment]
@tschottdorf Not to no avail. We're still experimenting here. Worth exploring that idea as well.

**Comment by petermattis** [comment]
LGTM
